contrib/kktix_bq_etl.py

```diff
@@ -0,0 +1,559 @@
+#!/usr/bin/env python3
+import argparse
+import hashlib
+import logging
+import re
+import unittest
+from typing import Dict, Set, List
+
+import pandas as pd
+import json
+from datetime import datetime
+from google.cloud import bigquery
+
+CANONICAL_COLUMN_NAMES_CORE = {
+    "paid_date",
+    "area_of_interest",
+    "payment_status",
+    "country_or_region",
+    "job_title",
+    "ticket_type",
+    "email",
+    "organization",
+    "price",
+    "dietary_habit",
+    "gender",
+    "years_of_using_python",
+    "registration_no",
+}
+
+
+CANONICAL_COLUMN_NAMES_2020_CORE = {
+    "ticket_type",
+    "payment_status",
+    "tags",
+    "paid_date",
+    "price",
+    "dietary_habit",
+    "years_of_using_python",
+    "area_of_interest",
+    "organization",
+    "job_title",
+    "country_or_region",
+    "departure_from_region",
+    "gender",
+    "email_from_sponsor",
+    "email_to_sponsor",
+    "ive_already_read_and_i_accept_the_epidemic_prevention_of_pycon_tw",
+    "ive_already_read_and_i_accept_the_privacy_policy_of_pycon_tw",
+    "email",
+    "registration_no",
+    "attendance_book",
+}
+
+CANONICAL_COLUMN_NAMES_2020_EXTRA_CORPORATE = {
+    "invoice_policy",
+    "invoiced_company_name",
+    "unified_business_no",
+    "pynight_attendee_numbers",
+    "know_financial_aid",
+    "have_you_ever_attended_pycon_tw",
+    "pynight_attending_or_not",
+    "how_did_you_know_pycon_tw",
+}
+
+CANONICAL_COLUMN_NAMES_2020_EXTRA_INDIVIDUAL = {
+    "pynight_attendee_numbers",
+    "know_financial_aid",
+    "have_you_ever_attended_pycon_tw",
+    "pynight_attending_or_not",
+    "how_did_you_know_pycon_tw",
+}
+
+CANONICAL_COLUMN_NAMES_2020_EXTRA_RESERVED: Set = set()
+
+
+CANONICAL_COLUMN_NAMES_2019_CORE = {
```

Comment:
seems to me that these `canonical_xxx` sets are unused?

Reply:
yes, the main code is from upload-kktix-ticket-csv-to-bigquery.py and I jus keep them, the set was used for unit test.